infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationMetadataManager.java

```diff
@@ -497,53 +497,68 @@ public void updateSnapshotSyncStatusOngoing(String clusterId, boolean forced, UU
     }
 
     /**
-     * Update replication status table's snapshot sync info as COMPLETED
-     * and update log entry sync status to ONGOING.
+     * Update replication status table's log entry sync as ONGOING.  Additionally, update the snapshot sync info as
+     * COMPLETED if updateSnapshotSyncInfo == true
      *
      * Note: TransactionAbortedException has been handled by upper level.
      *
      * @param clusterId standby cluster id
+     * @param remainingEntriesToSend An estimate of the number of entries remaining to be sent to standby cluster
+     * @param updateSnapshotSyncInfo boolean indicating if last snapshot sync's info must be updated
+     * @param baseSnapshot Corfu log timestamp at which the last snapshot was based
      */
-    public void updateSnapshotSyncStatusCompleted(String clusterId, long remainingEntriesToSend, long baseSnapshot) {
-        Instant time = Instant.now();
-        Timestamp timestamp = Timestamp.newBuilder().setSeconds(time.getEpochSecond())
-                .setNanos(time.getNano()).build();
+    public void setLogEntrySyncOngoing(String clusterId, long remainingEntriesToSend,
+                                       boolean updateSnapshotSyncInfo, long baseSnapshot) {
         ReplicationStatusKey key = ReplicationStatusKey.newBuilder().setClusterId(clusterId).build();
+        ReplicationStatusVal current;
 
         try (TxnContext txn = corfuStore.txn(NAMESPACE)) {
 
             CorfuStoreEntry<ReplicationStatusKey, ReplicationStatusVal, Message> record = txn.getRecord(replicationStatusTable, key);
 
-            if (record.getPayload() != null) {
-                ReplicationStatusVal previous = record.getPayload();
-                SnapshotSyncInfo previousSyncInfo = previous.getSnapshotSyncInfo();
-
-                SnapshotSyncInfo currentSyncInfo = previousSyncInfo.toBuilder()
-                        .setStatus(SyncStatus.COMPLETED)
-                        .setBaseSnapshot(baseSnapshot)
-                        .setCompletedTime(timestamp)
-                        .build();
-
-                ReplicationStatusVal current = ReplicationStatusVal.newBuilder()
-                        .setRemainingEntriesToSend(remainingEntriesToSend)
-                        .setSyncType(SyncType.LOG_ENTRY)
-                        .setStatus(SyncStatus.ONGOING)
-                        .setSnapshotSyncInfo(currentSyncInfo)
-                        .build();
-
-                txn.putRecord(replicationStatusTable, key, current, null);
-                txn.commit();
+            if (record.getPayload() == null) {
```

Comment:
yes.. added an error and a comment.  We do not need to commit the transaction as no operations have been done and TxnContext implements Autocloseable, which means it will invoke txn.close() when the method returns.